test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,394 @@
+                componentResized = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+                System.out.println("ComponentMoved: " + e.getSource());
+                componentMoved = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+                System.out.println("ComponentHidden: " + e.getSource());
+                componentHidden = true;
+                events.add(e);
+            }
+        };
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+
+            robot.mouseMove(
+                components[0].getLocationOnScreen().x
+                    + components[0].getSize().width / 2,
+                components[0].getLocationOnScreen().y
+                    + components[0].getSize().height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            // Hide all components and check if the ComponentEvent is triggered
+            for (int i = 0; i < components.length; i++) {
+                Component currentComponent = components[i];
+
+                resetValues();
+                EventQueue.invokeAndWait(() -> {
+                    currentComponent.setVisible(false);
+                    frame.invalidate();
+                    frame.validate();
```

Comment:
```
  frame.invalidate();
  frame.validate();
```
Can move these two into separate method and call it since its been used in several places.